### ISSUE TITLE
Strangeness in ColorExtensions.ToRgb method

### ISSUE THREAD

**Issue by gmoller** [craftworkgames/MonoGame.Extended#416]
I've been playing a bit with the Monogame particle system. I could not get the Color interpolation to work properly (interpolating from green to red). I've debugged it down to this odd line (in Monogame.Extended.ColorExtensions) line 30: 

    h = h/360f;

I guess the algorithm is taken from https://stackoverflow.com/questions/2353211/hsl-to-rgb-color-conversion (or similar), but in this algorithm there is no divided the hue by 360. If I remove this line and recompile, I seem to get my desired behaviour.

Do you know why it does this?

**Comment by gmoller** [comment]
Hi, ok I think I know where the problem is now. For my ColorInterpolator modifer I use: 
new ColorInterpolator
{
    Name = "Red->Green",
    StartValue = HslColor.FromRgb(Color.Red),
    EndValue = HslColor.FromRgb(Color.Lime)
}

as you can see uses the FromRgb static method of HslColor. The following test:
[Test]
public void FromRgbAndToRgbWorksCorrectly()
{
     HslColor blueHsl = HslColor.FromRgb(Color.Blue);
     Color blueRgb = blueHsl.ToRgb();

     Assert.AreEqual(Color.Blue, blueRgb);
 }
gives output:
Expected: {R:0 G:0 B:255 A:255}
But was:  {R:255 G:2 B:0 A:255}

I believe the FromRgb method is not working correctly. It normalizes the Hue to a value between 0 and 1. But I suspect it should be a value between 0 and 360 (which explains why the aforementioned ToRgb method divides by 360).

So, in a nutshell, the two methods (FromRgb and ToRgb) do not play well together.

**Comment by gmoller** [comment]
One more thing to note, the following tests work fine (which is why I don't the issue lies with ToRgb):
[TestCase(0f, 1f, 0.5f, "{R:255 G:0 B:0 A:255}")] // Color.Red
[TestCase(360f, 1f, 0.5f, "{R:255 G:0 B:0 A:255}")] // Color.Red
[TestCase(120f, 1f, 0.5f, "{R:0 G:255 B:0 A:255}")] // Color.Lime
public void ReturnsCorrectValue(float h, float s, float l, string expected)
{
    var hslColour = new HslColor(h, s, l);
    Color rgbColor = hslColour.ToRgb();
    Assert.AreEqual(expected, rgbColor.ToString());
}

**Comment by craftworkgames** [comment]
Hey mate

Yes, it looks like you've definitely uncovered a bug. The truth is, I haven't used this method a *lot* in my own stuff. I've noticed weird behavior before, but never really looked into it deeply. Thanks for investigating it.

Would you like to submit a PR with the fix?

**Comment by gmoller** [comment]
Sure thing. Took me a while to figure out how GitHub pull requests are done. Let me know if there is any problem.

**Comment by craftworkgames** [comment]
Looks good so far. Knowing how to do GitHub PR's is a good skill you have up your sleeve. Now that you've done it, you'll probably find more uses for said skill 😉

**Comment by gmoller** [comment]
Thanks. I'm used to the Microsoft world of Visual Studio where everything is spoon fed to me :).
One other thing I noticed, is that there are two unit tests failing: WhenThereAreNoActiveParticles_GracefullyDoesNothing and
WhenThereAreParticlesToExpire_DecreasesActiveParticleCount
I'd be interested to know if they work for you?

I can get them both to pass by setting the AutoTrigger flag on the emitter to false.

**Comment by craftworkgames** [comment]
> I'd be interested to know if they work for you?

Oh yeah, they are failing for me too. 

Although, I hadn't noticed because unfortunately I had to disable the unit tests on the build server fairly recently due to some errors I haven't figured out how to solve yet. The tests themselves seem fine (they run locally) but when they run with the test runner on the build server the wrong DLL is loaded for some reason I really don't understand yet.

Anyways, the `AutoTrigger` thing is also a recent addition, so I guess it makes sense that those tests broke.  I've been working on some improvements to the particle system. I'll try keep a better eye on the tests next time :)
